v2/sourcedb-to-spanner/README_Sourcedb_to_Spanner.md

```diff
@@ -121,49 +121,72 @@ need valid resources for the required parameters.
-export DATABASE_ID=<databaseId>
-export PROJECT_ID=<projectId>
-
-### Optional
-export DRIVER_JARS=<driverJars>
-export DRIVER_CLASS_NAME=<driverClassName>
-export CONNECTION_URL=<connectionURL>
-export CONNECTION_PROPERTIES=<connectionProperties>
-export USERNAME=<username>
-export PASSWORD=<password>
-export TABLES=<tables>
-export NUM_PARTITIONS=<numPartitions>
-export SPANNER_HOST=https://batch-spanner.googleapis.com
-export IGNORE_COLUMNS=<ignoreColumns>
-export DISABLED_ALGORITHMS=<disabledAlgorithms>
-export EXTRA_FILES_TO_STAGE=<extraFilesToStage>
-
-gcloud dataflow flex-template run "sourcedb-to-spanner-flex-job" \
-  --project "$PROJECT" \
-  --region "$REGION" \
-  --template-file-gcs-location "$TEMPLATE_SPEC_GCSPATH" \
-  --parameters "jdbcDriverJars=$DRIVER_JARS" \
-  --parameters "jdbcDriverClassName=$DRIVER_CLASS_NAME" \
-  --parameters "sourceConnectionURL=$CONNECTION_URL" \
-  --parameters "sourceConnectionProperties=$CONNECTION_PROPERTIES" \
-  --parameters "username=$USERNAME" \
-  --parameters "password=$PASSWORD" \
-  --parameters "tables=$TABLES" \
-  --parameters "numPartitions=$NUM_PARTITIONS" \
-  --parameters "instanceId=$INSTANCE_ID" \
-  --parameters "databaseId=$DATABASE_ID" \
-  --parameters "projectId=$PROJECT_ID" \
-  --parameters "spannerHost=$SPANNER_HOST" \
-  --parameters "ignoreColumns=$IGNORE_COLUMNS" \
-  --parameters "disabledAlgorithms=$DISABLED_ALGORITHMS" \
-  --parameters "extraFilesToStage=$EXTRA_FILES_TO_STAGE"
+# Adding timestamp just allows the name to be unique across multiple runs
+JOB_NAME="<job-name>-`date +'%Y-%m-%d-%H-%M-%S-%N'`"
+TEMPLATE_NAME="Sourcedb_to_Spanner_Flex"
+PROJECT="<GCP-PROJECT-NAME>"
+
+# Note this changes the default project of your bash session.
+gcloud config set project $PROJECT
+
+# Below are GCS paths, they could be in same bucket separated by folders.
+TEMPLATE_BUCKET_NAME="<GCS_BUCKET_WHERE_TEMPLATE_IS_BUILT>"
+TEMPLATE_PATH="${TEMPLATE_BUCKET_NAME}/templates/flex/${TEMPLATE_NAME}"
+DLQ_DIRECTORY="<GCS_PATH_FOR_DLQ>"
+
+# Source Parameters
+SRC_USER_ID="<SOURCE_DB_USER_ID>"
+SRC_PWD="<SOURCE_PASSWORD>"
+SRC_IP="<SOURCE_ID>"
+SOURCE_HOST="${SRC_IP}"
+SOURCE_PORT="3306"
+SOURCE_DB="<NAME_OF_SOURCE_DB>"
+
+# Spanner details
+SPANNER_INSTANCE_ID="<SPANNER_INSTANCE_ID>"
+SPANNER_DB_ID="<SPANNER_DB_ID>"
+
+
+# Dataflow Parameters
+WORKER_REGION="<REGION_TO_SPAWN_DATAFLOW_WORKERS like us-central1>"
+NETWORK="<VPC network to spawn dataflow workers>"
+SUBNETWORK="regions/${WORKER_REGION}/subnetworks/<SUBNET_NAME>"
+
+# Typical Configuration for dataflow scaling, please change as per load
+MAX_WORKERS=10
+NUM_WORKERS=10
+WORKER_MACHINE_TYPE="n1-highmem-32"
+# Override NUM_PARTITIONS for tall tables.
+# NUM_PARTITIONS="4000"
+
+set -x;
+date
+curl -H 'Content-type: application/json' -H "Authorization: Bearer $(gcloud auth print-access-token)" -XPOST "https://dataflow.googleapis.com/v1b3/projects/${PROJECT}/locations/${WORKER_REGION}/flexTemplates:launch" -d"{
```

Comment:
curl is alternative to `flex-template run` we can have either of them.

Reply:
updated from the auto gen docs.